components/camel-arangodb/src/main/java/org/apache/camel/component/arangodb/ArangoDbProducer.java

```diff
@@ -116,7 +116,7 @@ private Function<Exchange, Object> saveDocument() {
                 Boolean isMultiInsert = (Boolean) exchange.getMessage().getHeader(MULTI_INSERT, false);
```

Comment:
getHeader(x,y, Boolean.class) would then avoid boxing also

Reply:
Yeah, that's even better.